File: java-sdk/src/main/java/com/owiseman/dataapi/sdk/auth/AuthenticationProvider.java
package com.owiseman.dataapi.sdk.auth;

import com.owiseman.dataapi.sdk.exception.AuthenticationException;

/**
 * 认证提供者接口
 */
public interface AuthenticationProvider {
    
    /**
     * 获取访问令牌
     * @return 访问令牌
     * @throws AuthenticationException 认证异常
     */
    String getAccessToken() throws AuthenticationException;
    
    /**
     * 刷新访问令牌
     * @return 新的访问令牌
     * @throws AuthenticationException 认证异常
     */
    String refreshToken() throws AuthenticationException;
    
    /**
     * 检查令牌是否有效
     * @return true如果令牌有效，否则false
     */
    boolean isTokenValid();
    
    /**
     * 清除认证信息
     */
    void clearAuthentication();
    
    /**
     * 获取认证类型
     * @return 认证类型
     */
    AuthenticationType getAuthenticationType();
    
    /**
     * 认证类型枚举
     */
    enum AuthenticationType {
        BEARER_TOKEN,
        API_KEY,
        BASIC_AUTH,
        OAUTH2
    }
}

File: typescript-sdk/src/auth/AuthenticationProvider.ts
import { AuthenticationType, TokenResponse } from '../types';

/**
 * 认证提供者接口
 */
export interface AuthenticationProvider {
  /**
   * 获取认证类型
   */
  getAuthenticationType(): AuthenticationType;
  
  /**
   * 获取访问令牌
   */
  getAccessToken(): Promise<string>;
  
  /**
   * 刷新访问令牌
   */
  refreshToken(): Promise<TokenResponse>;
  
  /**
   * 验证令牌是否有效
   */
  isTokenValid(): Promise<boolean>;
  
  /**
   * 清除认证信息
   */
  clearAuthentication(): void;
  
  /**
   * 获取认证头
   */
  getAuthenticationHeaders(): Promise<Record<string, string>>;
}